src/react/projects/spark-core-react/src/SprkPromo/SprkPromo.test.js

```diff
@@ -0,0 +1,359 @@
+  const wrapper = mount( <SprkPromo
+    isFlag 
+    imgLinkHref='foo' 
+    imgLinkAnalytics='bar' 
+    imgLinkIdString='baz' 
+    imgSrc='foobar'
+  />);
+  const link = wrapper.find('a.sprk-b-Link--plain');
+  
+  expect(link.length).toBe(1);
+  expect(link.find('[href="foo"]').length).toBe(1);
+  expect(link.find('[data-analytics="bar"]').length).toBe(1);
+  expect(link.find('[data-id="baz"]').length).toBe(1);
+});
+
+it ('should render a link with the correct classes when isFlag and additionalClassesImgLink are provided', () => {
+  const wrapper = mount( <SprkPromo
+    isFlag 
+    additionalClassesImgLink = 'bobsClass'
+    imgSrc = 'foobar'
+  />);
+  const link = wrapper.find('a.sprk-b-Link--plain');
+  
+  expect(link.length).toBe(1);
+  expect(link.hasClass('sprk-o-Stack__item--fourth@s')).toBe(true);
+  expect(link.hasClass('sprk-o-Stack__item')).toBe(true);
+  expect(link.hasClass('bobsClass')).toBe(true);
+});
+
+// Non-flag image and link 
+
+it ('should render an image with passed in properties', () => {
+  const wrapper = shallow(<SprkPromo imgAlt='foo' imgSrc='bar' />);
+  const img = wrapper.find('img.sprk-c-Promo__image');
+  expect(img.length).toBe(1);
+  expect(img.find('[alt="foo"]').length).toBe(1);
+  expect(img.find('[src="bar"]').length).toBe(1);
+});
+
+it ('should not render an image if no src is provided', () => {
+  const wrapper = shallow( <SprkPromo />);
+  
+  expect(wrapper.find('img').length).toBe(0);
+});
+
+it ('should render an image link with the correct properties', () => {
+  const wrapper = mount( <SprkPromo
+    imgLinkHref='foo' 
+    imgLinkAnalytics='bar' 
+    imgLinkIdString='baz' 
+    imgSrc='foobar'
+  />);
+  const link = wrapper.find('a.sprk-b-Link--plain');
+  
+  expect(link.length).toBe(1);
+  expect(link.find('[href="foo"]').length).toBe(1);
+  expect(link.find('[data-analytics="bar"]').length).toBe(1);
+  expect(link.find('[data-id="baz"]').length).toBe(1);
+});
+
+it ('should render a link with the correct classes when imgSrc and additionalClassesImgLink are provided', () => {
+  const wrapper = mount( <SprkPromo
+    additionalClassesImgLink = 'bobsClass'
+    imgSrc = 'foobar'
+  />);
+  const link = wrapper.find('a.sprk-b-Link--plain');
+  
+  expect(link.length).toBe(1);
+  expect(link.hasClass('sprk-o-Stack__item--half@s')).toBe(true);
+  expect(link.hasClass('sprk-o-Stack__item')).toBe(true);
+  expect(link.hasClass('bobsClass')).toBe(true);
+});
+
+it ('should render a link with the correct classes when additionalClassesImgLink are provided', () => {
+  const wrapper = mount( <SprkPromo
+    imgSrc = 'foobar'
+    additionalClassesImgLink = 'bobsClass'
+  />);
+  const link = wrapper.find('a.sprk-b-Link--plain');
+  
+  expect(link.length).toBe(1);
+  expect(link.hasClass('sprk-o-Stack__item--half@s')).toBe(true);
+  expect(link.hasClass('sprk-o-Stack__item')).toBe(true);
+  expect(link.hasClass('bobsClass')).toBe(true);
+});
+
+// Body elements
+
```

Comment:
Noticed you are using JS comments to denote the different sections. I like that, but, If you wanted to output the section when the tests run, you could wrap each test section in `describe('Promo Body Elements', () => {it (...); );`